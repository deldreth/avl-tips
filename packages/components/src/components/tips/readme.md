# avl-tips



<!-- Auto Generated Below -->


## Dependencies

### Depends on

- [avl-tips-card](../card)

### Graph
```mermaid
graph TD;
  avl-tips --> avl-tips-card
  style avl-tips fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
