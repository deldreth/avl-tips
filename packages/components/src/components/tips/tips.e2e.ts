import { newE2EPage } from '@stencil/core/testing';

describe('avl-tips', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<avl-tips></avl-tips>');

    const element = await page.find('avl-tips');
    expect(element).toHaveClass('hydrated');
  });
});
