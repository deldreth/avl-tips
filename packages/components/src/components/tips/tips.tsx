import { Component, Host, State, h } from "@stencil/core";
import axios from "axios";
import config from "../../config"

type Employee = {
  name: string;
  venmo: string;
  cash: string;
  employer: string;
};

@Component({
  tag: "avl-tips",
  styleUrl: "tips.css",
  shadow: true
})
export class Tips {
  @State() employee: Employee;
  @State() hasError: boolean;

  componentWillLoad() {
    axios
      .get(
        `${config.api.prod.url}/employee`
      )
      .then(response => (this.employee = response.data))
      .catch(() => (this.hasError = true));
  }

  render() {
    return (
      <Host>
        <avl-tips-card {...this.employee}></avl-tips-card>
      </Host>
    );
  }
}
