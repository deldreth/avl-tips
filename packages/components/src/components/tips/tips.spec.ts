import { newSpecPage } from "@stencil/core/testing";
import { Tips } from "./tips";
import mockAxios from "jest-mock-axios";

describe("avl-tips", () => {
  afterEach(() => {
    mockAxios.reset();
  });

  it("matches snapshot", async () => {
    const page = await newSpecPage({
      components: [Tips],
      html: "<avl-tips></avl-tips>"
    });

    expect(page.root).toMatchSnapshot();
  });

  it("componentWillLoad sets state", () => {
    const tips = new Tips();
    const mockEmployee = {
      name: "Test Name",
      venmo: "Venmo",
      cash: "Cash",
      paypal: "Paypal",
      employer: "Employer"
    };

    tips.componentWillLoad();
    mockAxios.mockResponse({
      data: mockEmployee
    });

    expect(tips.employee).toEqual(mockEmployee);
  });
});
