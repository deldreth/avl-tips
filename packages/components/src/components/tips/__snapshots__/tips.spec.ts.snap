// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`avl-tips matches snapshot 1`] = `
<avl-tips>
  <mock:shadow-root>
    <avl-tips-card></avl-tips-card>
  </mock:shadow-root>
</avl-tips>
`;
